StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/MemberOrderHelper.cs

```diff
@@ -0,0 +1,124 @@
+namespace StyleCop.Analyzers.OrderingRules
+{
+    using System;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using StyleCop.Analyzers.Helpers;
+
+    /// <summary>
+    /// Helper class for dealing with member priority.
+    /// </summary>
+    public class MemberOrderHelper
+    {
+        private readonly ModifierFlags modifierFlags;
+        private readonly Accessibility accessibilty;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MemberOrderHelper"/> class.
+        /// </summary>
+        /// <param name="member">The member to wrap.</param>
+        public MemberOrderHelper(MemberDeclarationSyntax member)
+        {
+            this.Member = member;
+            var modifiers = member.GetModifiers();
+
+            this.modifierFlags = GetModifierFlags(modifiers);
+            this.accessibilty = GetAccessibilityFlags(modifiers);
+        }
+
+        [Flags]
+        private enum ModifierFlags
+        {
+            /// <summary>
+            /// No modifiers
+            /// </summary>
+            None = 0,
+            /// <summary>
+            /// Readonly modifier
+            /// </summary>
+            Readonly = 1,
+            /// <summary>
+            /// Static modifier
+            /// </summary>
+            Static = 1 << 2,
+            /// <summary>
+            /// Const modifier
+            /// </summary>
+            Const = 1 << 3,
+        }
+
+        /// <summary>
+        /// The wrapped member.
+        /// </summary>
+        /// <value>
+        /// The wrapped member.
+        /// </value>
+        public MemberDeclarationSyntax Member { get; }
+
+        /// <summary>
+        /// The priority for this member.
+        /// </summary>
+        /// <value>
+        /// The priority for this member.
+        /// </value>
+        public int Priority => (int)this.modifierFlags + ((int)this.accessibilty * 100/*the * 100 ensures the accesibility is more important than the modifier*/);
+
+        /// <summary>
+        /// Checks if the current field should be declared before the other.
+        /// </summary>
+        /// <param name="other">The field to compare against.</param>
+        /// <returns>True if the field should be delcared before the other.</returns>
+        public bool ShouldBeBefore(MemberOrderHelper other) => this.Priority > other.Priority;
+
+        private static Accessibility GetAccessibilityFlags(SyntaxTokenList syntax)
```

Comment:
@sharwell Not sure I agree with using `Microsoft.CodeAnalysis.Accessibility`. Below are it's values which do not match the order of StyleCop.

```
NotApplicable = 0
Private = 1
ProtectedAndInternal = 2
Protected = 3
Internal = 4
ProtectedOrInternal = 5
Public = 6
```

Ending up making `AccessLevel` and `AccessLevelHelper` in `StyleCop.Analyzers.Helpers` for determining access levels. `AccessLevelHelper.GetAccessLevel` could be used in place of this method.